includes/services/DuplicationManager.php

```diff
@@ -0,0 +1,353 @@
+<?php
+
+namespace YesWiki\Core\Service;
+
+use Exception;
+use YesWiki\Bazar\Field\FileField;
+use YesWiki\Bazar\Field\ImageField;
+use YesWiki\Bazar\Field\TextareaField;
+use YesWiki\Bazar\Service\FormManager;
+use YesWiki\Bazar\Service\EntryManager;
+use YesWiki\Bazar\Service\ListManager;
+use YesWiki\Core\Service\PageManager;
+use YesWiki\Wiki;
+
+class DuplicationManager
+{
+    protected $uploadPath;
+    protected $wiki;
+
+
+    /**
+     * DuplicationManager constructor
+     * @param Wiki $wiki the injected Wiki instance
+     */
+    public function __construct(Wiki $wiki)
+    {
+        $this->wiki = $wiki;
+        $this->uploadPath = $this->getLocalFileUploadPath();
+    }
+
+    /**
+     * Get the local path to files uploads (usually "files")
+     *
+     * @return string local path to files uploads
+     */
+    private function getLocalFileUploadPath()
+    {
+        $attachConfig = $this->wiki->config['attach_config'];
+
+        if (!is_array($attachConfig)) {
+            $attachConfig = array();
+        }
+
+        if (empty($attachConfig['upload_path'])) {
+            $this->uploadPath = 'files';
+        } else {
+            $this->uploadPath = $attachConfig['upload_path'];
+        }
+
+        return $this->uploadPath;
+    }
+
+    /**
+     * Return fields that may contain attachments to import (fichier, image, or textelong fields for bazar entries)
+     *
+     * @param array $id
+     * @return array keys of fields that may contain attachments to import
+     */
+    public function getUploadFieldsFromEntry($id)
+    {
+        $fields = [];
+        $entry = $this->wiki->services->get(EntryManager::class)->getOne($id);
+        if (!empty($entry['id_fiche'])) { // bazar entry
+            $formManager = $this->wiki->services->get(FormManager::class);
+            $form = $formManager->getOne($entry['id_typeannonce']);
+            // find fields that are textareas
+            foreach ($form['prepared'] as $field) {
+                if ($field instanceof TextareaField or $field instanceof ImageField or $field instanceof FileField) {
+                    $fields[] = [
```

Comment:
je pense qu'il vaudrait mieux retourner le $field entrier, plutot que ce tableau avec juste ID et TYPE, comme ça on sait ce que contient la variable juste avec le nom (un $field c'est un BazarField, pas un array [id, type]) et aussi plus loin dans le code ça évite des check un foireux du genre $field[type] == 'image', c'est mieux de refaire un $field instance of ImageField